aliyun/src/test/java/org/apache/iceberg/aliyun/oss/mock/AliyunOSSMockRule.java

```diff
@@ -0,0 +1,143 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.aliyun.oss.mock;
+
+import com.aliyun.oss.OSS;
+import com.aliyun.oss.OSSClientBuilder;
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.util.Map;
+import org.apache.commons.io.FileUtils;
+import org.apache.iceberg.aliyun.oss.AliyunOSSTestRule;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AliyunOSSMockRule implements AliyunOSSTestRule {
+  private static final Logger LOG = LoggerFactory.getLogger(AliyunOSSMockRule.class);
+
+  private final Map<String, Object> properties;
+
+  private AliyunOSSMockApp ossMockApp;
+
+  private AliyunOSSMockRule(Map<String, Object> properties) {
+    this.properties = properties;
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  @Override
+  public void start() {
+    ossMockApp = AliyunOSSMockApp.start(properties);
+  }
+
+  @Override
+  public void stop() {
+    ossMockApp.stop();
+  }
+
+  @Override
+  public OSS createOSSClient() {
+    String endpoint = String.format("http://localhost:%s", properties.getOrDefault(
+        AliyunOSSMockApp.PROP_HTTP_PORT,
+        AliyunOSSMockApp.PORT_HTTP_PORT_DEFAULT));
+    return new OSSClientBuilder().build(endpoint, "foo", "bar");
+  }
+
+  @Override
+  public String keyPrefix() {
+    return "mock-objects/";
+  }
+
+  private File rootDir() {
+    Object rootDir = properties.get(AliyunOSSMockApp.PROP_ROOT_DIR);
+    Preconditions.checkNotNull(rootDir, "Root directory cannot be null");
+    return new File(rootDir.toString());
+  }
+
+  @Override
+  public void setUpBucket(String bucket) {
+    createOSSClient().createBucket(bucket);
+  }
+
+  @Override
+  public void tearDownBucket(String bucket) {
+    try {
+      Files.walk(rootDir().toPath())
+          .filter(p -> p.toFile().isFile())
+          .forEach(p -> {
+            try {
+              Files.delete(p);
+            } catch (IOException e) {
+              // delete this files quietly.
+            }
+          });
+
+      createOSSClient().deleteBucket(bucket);
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+  }
+
+  public static class Builder {
+    private Map<String, Object> props = Maps.newHashMap();
+
+    public Builder withRootDir(String rootDir) {
+      props.put(AliyunOSSMockApp.PROP_ROOT_DIR, rootDir);
+      return this;
+    }
+
+    public Builder silent() {
+      props.put(AliyunOSSMockApp.PROP_SILENT, true);
+      return this;
+    }
+
+    public AliyunOSSMockRule build() {
+      if (props.get(AliyunOSSMockApp.PROP_ROOT_DIR) == null) {
```

Comment:
So when the PROP_ROOT_DIR is not set or set to empty string,  we will use the default temporary directory with a current timestamp suffix.  Then we will create the directory by using the `createRootDir` method .  I think it's more clear to decide which root dir value we will use firstly, then check whether it's created for testing: 

```java
    public AliyunOSSMockRule build() {
      String rootDir = (String) props.get(AliyunOSSMockApp.PROP_ROOT_DIR);
      if (StringUtils.isNullOrEmpty(rootDir)) {
        File dir = new File(FileUtils.getTempDirectory(), "oss-mock-file-store" + System.currentTimeMillis());
        rootDir = dir.getAbsolutePath();
      }
      File root = new File(rootDir);
      root.deleteOnExit();
      root.mkdir();

      return new AliyunOSSMockRule(props);
    }
```

Reply:
Maybe we will still need to set the rootDir back to the properties Map<String, Object> in the above code.  I'm trying not to access the PROP_ROOT_DIR again and again in the current PR.

Reply:
yes, we need set the rootDir only if no such element in the properties Map.